logstash-core/lib/logstash/config/defaults.rb

```diff
@@ -6,6 +6,14 @@ module LogStash module Config module Defaults
 
   extend self
 
+  def input
```

Comment:
Well, the main reason for constants is performance / memory allocation, and this will never be called more than a couple times per invocation so I don't think the reasoning behind using constants in general applies here.